### PR TITLE
starting confirmation page after room creation

### PR DESCRIPTION
**Pull request by ntchou**
- call created confirmation page (button to open the call + copy-to-clipboard share link)
- the API sends back `roomUrl` on `createRoom` call
- fix local join invitation link (when using emulator)
- move `CreateRoomForm` into its own directory
- extract utility functions from `CreateRoomForm`

### CODE REVIEW

**Review by HugoGresse — APPROVED** [review]

File: functions/src/firebase/env.ts [hunk 1]
```diff
@@ -34,15 +35,23 @@ export const getTwilioEnv = (): TwilioEnv => {
...
     } = functions.config()
     if (!domain || !emailfrom) {
         throw new InternalServerError('Missing app domain or emailfrom env')
     }
+
+    if (isUsingEmulator()) {
+        domain = 'localhost:3000'
+    }
+
+    const protocol = isUsingEmulator() ? 'http' : 'https'
```
Comment (HugoGresse) [inline functions/src/firebase/env.ts:49]: .runtimeconfig contain local env, this protocol could be added there as well as setting the localhost domain:port.

File: src/pages/AdminDashboard/AdminDashboard.tsx [hunk 2]
```diff
@@ -45,6 +47,10 @@ const AdminDashboard = () => {
             getUserDetails()
             dispatch(getRooms(t))
         }
+
+        // return () => {
+        //     dispatch(newRoom())
```
Comment (HugoGresse) [inline src/pages/AdminDashboard/AdminDashboard.tsx:52]: remove comment?

File: src/pages/AdminDashboard/CreateRoomForm/CreateRoomForm.tsx [hunk 3]
```diff
@@ -237,13 +173,7 @@ const CreateRoomForm = ({ fields, onFormSubmit, onCreateFormReset }) => {
...
             enableReinitialize
             initialValues={fields}
-            validate={(values) => {
-                const errors: Partial<Room> = {}
-                if (!values.name) {
-                    errors.name = 'Required'
-                }
-                return errors
-            }}
+            //validationSchema={validationSchema}
```
Comment (HugoGresse) [inline src/pages/AdminDashboard/CreateRoomForm/CreateRoomForm.tsx:176]: comment needed?

File: src/pages/AdminDashboard/CreateRoomForm/createRoomTools.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        case UNITS.days:
+            remindBeforeTimestampSecs = notification.number * 60 * 60 * 24
+            break
+    }
+
+    return startAt - remindBeforeTimestampSecs
+}
+
+export const validateEmail = (email) => {
+    const regexp = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```
Comment (HugoGresse) [inline src/pages/AdminDashboard/CreateRoomForm/createRoomTools.ts:32]: with yup: `string.required().email()`

File: src/pages/AdminDashboard/CreateRoomForm/createRoomTools.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    return isValid
+}
+
+export const isParticipantValid = (participant) => {
+    return validateEmail(participant) || validatePhoneNumber(participant)
+}
+
+export const isNumeric = (destination) => {
+    for (let char of destination) {
+        if (char >= '0' && char <= '9') continue
```
Comment (HugoGresse) [inline src/pages/AdminDashboard/CreateRoomForm/createRoomTools.ts:54]: what if the first char is a "+"?

File: src/pages/AdminDashboard/roomsActions.ts [hunk 6]
```diff
@@ -36,16 +58,22 @@ export const createRoom = (t, room: Room, remindAt: number) => async (
     const api = new Api(token)
 
     try {
-        const { roomId } = await api.createRoom(room)
+        const { roomId, roomUrl } = await api.createRoom(room)
         dispatch(getRooms(t))
-        dispatch(createReminder(t, roomId, remindAt))
+
+        if (remindAt) dispatch(createReminder(t, roomId, remindAt))
```
Comment (HugoGresse) [inline src/pages/AdminDashboard/roomsActions.ts:64]: createRoom can also be passed 3 params to create reminders: destinations, sendsAt array, and hostName

**Review by HugoGresse — CHANGES_REQUESTED** [review]

File: functions/src/firebase/env.ts (hunk 1 above)
Comment (MattiaPrimavera) [reply functions/src/firebase/env.ts:49]: this way wouldn't we need a separate `.runtimeConfig` file for `dev` & `local` environments ?

File: src/pages/AdminDashboard/CreateRoomForm/CreateRoomForm.tsx (hunk 3 above)
Comment (MattiaPrimavera) [reply src/pages/AdminDashboard/CreateRoomForm/CreateRoomForm.tsx:173]: @ntchou will take on with form validation later

File: src/pages/AdminDashboard/CreateRoomForm/createRoomTools.ts (hunk 4 above)
Comment (MattiaPrimavera) [reply src/pages/AdminDashboard/CreateRoomForm/createRoomTools.ts:32]: same as above

File: src/pages/AdminDashboard/CreateRoomForm/createRoomTools.ts (hunk 5 above)
Comment (MattiaPrimavera) [reply src/pages/AdminDashboard/CreateRoomForm/createRoomTools.ts:54]: validation is currently commented in order to be replaced with the use of `yup`

File: src/pages/AdminDashboard/AdminDashboard.tsx (hunk 2 above)
Comment (ntchou) [reply src/pages/AdminDashboard/AdminDashboard.tsx]: why did you comment. If you change page you keep previous state and if you come back on admin dashboard you'll have previous message displayed

File: src/pages/AdminDashboard/roomsActions.ts (hunk 6 above)
Comment (MattiaPrimavera) [reply src/pages/AdminDashboard/roomsActions.ts:64]: #477

File: src/pages/AdminDashboard/CreateRoomForm/createRoomTools.ts (hunk 5 above)
Comment (ntchou) [reply src/pages/AdminDashboard/CreateRoomForm/createRoomTools.ts:54]: i had already ask here and gave a suggestion https://github.com/Instant-Visio/InstantVisio-WebApp/pull/455#discussion_r550265953

**Comment by ntchou** [comment]
thx @MattiaPrimavera

File: src/pages/AdminDashboard/AdminDashboard.tsx (hunk 2 above)
Comment (MattiaPrimavera) [reply src/pages/AdminDashboard/AdminDashboard.tsx]: uncommented already, it was a push mistake

File: src/pages/AdminDashboard/CreateRoomForm/CreateRoomForm.tsx (hunk 3 above)
Comment (ntchou) [reply src/pages/AdminDashboard/CreateRoomForm/CreateRoomForm.tsx:173]: i did some tests and i forgot

**Pull request merged**